### PR TITLE
feat(native): Merge source names into symbolicator responses

### PR DESCRIPTION
**Pull request by jan-auer**
Symbolicator has a limited concept of sources compared to Sentry. Most notably,
it does not know about names assigned to sources in settings. This attaches the
additional source name to the symbolication response so that the UI can render
it conveniently with the list of candidates.

In follow-up work, the list of candidates will be trimmed at this spot.

### CODE REVIEW

**Comment by jan-auer** [comment]
cc @priscilawebdev. The new field is called `source_name`.

@untitaker beware that this is yet untested.

**Comment by jan-auer** [comment]
Verified that this works. We don't have comprehensive tests for candidates yet, so we can amend those at a later point.

File: src/sentry/lang/native/symbolicator.py [hunk 1]
```diff
@@ -360,6 +361,18 @@ def _ensure_open(self):
         if not self.session:
             raise RuntimeError("Session not opened")
 
+    def _process_response(self, json):
+        source_names = {source["id"]: source.get("name") for source in self.sources}
+        source_names[INTERNAL_SOURCE_NAME] = "Sentry"
+
+        for module in json.get("modules") or ():
+            for candidate in module.get("candidates") or ():
+                candidate["source_name"] = (
```
Comment (jan-auer) [inline src/sentry/lang/native/symbolicator.py:370]: This is aligned with: 

https://github.com/getsentry/sentry/blob/71fcb6d6165060c06e9d763f868a8c4770eba91b/src%2Fsentry%2Fstatic%2Fsentry%2Fapp%2Ftypes%2FdebugImage.tsx#L62-L67

File: src/sentry/lang/native/symbolicator.py [hunk 2]
```diff
@@ -360,6 +361,18 @@ def _ensure_open(self):
...
+                    source_names.get(candidate["source"]) or "Unnamed Repository"
```
Comment (jan-auer) [inline src/sentry/lang/native/symbolicator.py:371]: The fallback of "Unnamed Repository" is aligned with the settings URL:

https://github.com/getsentry/sentry/blob/b8bf85948d4c4832c60ab31e9f1a5a3c39fd53cf/src%2Fsentry%2Fstatic%2Fsentry%2Fapp%2Fdata%2Fforms%2FprojectDebugFiles.jsx#L112

I'm realizing that this is something we should be doing in the UI, however.

Comment (jan-auer) [reply src/sentry/lang/native/symbolicator.py]: @priscilawebdev I think using the same `<em>{t('<Unnamed Repository>')}</em>` in the UI for candidates without a `source_name` would be the best option here.

**Review by untitaker — APPROVED** [review]

**Pull request merged**